external/js/cothority/src/byzcoin/client-transaction.ts

```diff
@@ -320,6 +323,88 @@ export class Instruction extends Message<Instruction> {
         return this.deriveId(what);
     }
 
+    /**
+     * Return a general-purpose json representation of an instruction that is
+     * contracts aware, ie. knows how to render specific contracts arguments. In
+     * the case a problem occurs (for example the requested contract does not
+     * implement a beautify function), the status property is set to non-zero.
+     *
+     * The JSON has the following structure:
+     *
+     *  {
+     *      status: 0 | 1
```

Comment:
What is the status here? Shouldn't it just throw an error in case of error?

Reply:
For the moment non-zero means there is an error, and we only use 1 but we might add additional error code.
I find it more convenient to use than throwing an error.

Reply:
Hmm - thinking out loud (or whatever applies when writing text):

Is it an error if `beautify` cannot return a nice representation?

Perhaps one solution would be to include a default representation which only sets the type-string and iterates over all arguments and sets the `value` to `key:value.toString('hex')`. Then I think the status would make sense: 
- `0` means only the default representation is returned
- `1` means it has been interpreted specifically
- `2` means it also has links (or whatever)